src/libs/WorkspacesSettingsUtils.ts

```diff
@@ -157,14 +158,14 @@ function checkIfWorkspaceSettingsTabHasRBR(policyID?: string) {
 /**
  * @returns a map where the keys are policyIDs and the values are BrickRoads for each policy
  */
-function getWorkspacesBrickRoads(): Record<string, BrickRoad> {
-    if (!allReports) {
+function getWorkspacesBrickRoads(reports: OnyxCollection<Report>, policies: OnyxCollection<Policy>, reportActions?: OnyxCollection<ReportActions>): Record<string, BrickRoad> {
```

Comment:
All good ?

Reply:
> which is already what we did before, so this param is useles

Not quite, you can find more details in the [proposal's](https://github.com/Expensify/App/issues/39928#issuecomment-2045740587) RCA / alternative solution, as to why we're not using the local `allReports` and instead passing it down from the component `reports` coming from the `useOnyx` hook.

**TLDR:** Because local action file's `allReports` is not hooked to the component's lifecycle which causes a delay in showing the workspace switcher data when refreshing the page (see issue and selected proposal). Passing down `reports` from the component via `useOnyx` makes it such that upon refresh the UI is instant (no delays).

Reply:
> same for reportActions, do we need to pass them down or should we subscribe to them here instead

I did this (additionally from my proposal) because on mWeb the RBR/GRB wouldn't show up in workspace switcher upon page refresh based on local action file's `reportActionsByReport`.

Reply:
Oh I see.... so isn't this same problem going to happen with basically any component using the methods on this file that use `allReports`  and any other collection we subscribe here that also has `waitForCollectionCallback` set to true?

Reply:
As of now `getWorkspacesBrickRoads` and `getWorkspacesUnreadStatuses` are only used within the `WorkspaceSwitcherPage` component.

It is true that if these functions were to be used in other components and keep the external `Onyx.connect` data implementation, that could cause similar stale data problems as it happened with the issue this PR fixes.

If you look at the issue's proposals, you'll see the differences in UI between the other proposals and why my alternative solution using `useOnyx` was selected (UI is instant).

I think the changes from this PR will make it such that for the future, if these functions are used in other components it will enforce similar usage by passing `reports`, `policies` and optionally `reportActions` straight from the component (lifecycle) either from component's props (`withOnyx` HOC) or from the recent `useOnyx` hook implementation.

Reply:
> I think the changes from this PR will make it such that for the future, if these functions are used in other components it will enforce similar usage by passing reports, policies and optionally reportActions straight from the component

I am confused, since this PR makes the param optional, people can misuse it by not passing the param. So if this bug will show up for anyone not passing the param, we should remove its optionality.

Reply:
Indeed, makes more sense, to have the `getWorkspacesBrickRoads`'s `reportActions` **required** but keep the `getBrickRoadForPolicy`'s `altReportActions` **optional** since `getBrickRoadForPolicy` is also used here:

https://github.com/Expensify/App/blob/26c9eba8c209090b5c17f6ba8c5a6c318555ad05/src/libs/WorkspacesSettingsUtils.ts#L113

in the `getChatTabBrickRoad` function which we don't want to change, keep it backwards compatible.

cc @iwiznia